### ISSUE TITLE
Send refactoring PR for TileLayerContainer.js

### ISSUE THREAD

**Issue by leonardehrenfried** [mfdz/digitransit-ui#164]
We have to touch the `TileLayerContainer.js` every now and again and every time it's difficult to understand the code.

This file should be refactored upstream and then merged into out repo through the normal downstreaming process.

Problems with the code

- `this.state.selectableTargets[0].layer` is repeated over and over. Please put that into a variable.
- The render method is huge. Please break it down into separate methods and give them a nice name.

**Comment by Beck-berry** [comment]
Check out commit 26151166 then I'll raise a PR.

**Comment by leonardehrenfried** [comment]
Can you also put these three sections into separate methods so you one can figure out what these parts of the code are there for? (Took me a while to understand it.)

- https://github.com/mfdz/digitransit-ui/blob/26151166c7bbc81a88dd2f261915d3bd2db08633/app/component/map/tile-layer/TileLayerContainer.js#L332-L358
- https://github.com/mfdz/digitransit-ui/blob/26151166c7bbc81a88dd2f261915d3bd2db08633/app/component/map/tile-layer/TileLayerContainer.js#L360-L373
- https://github.com/mfdz/digitransit-ui/blob/26151166c7bbc81a88dd2f261915d3bd2db08633/app/component/map/tile-layer/TileLayerContainer.js#L375-L388

So in other words, the `render` method should just be a short method that lists all the possible code paths but the actual implementation is delegated to individual methods.

**Comment by Beck-berry** [comment]
Ok, next commit: 6a8ab4c0. If you find it good enough, I'll squash the two.

**Comment by leonardehrenfried** [comment]
Yes, looks good but please put `this.state.selectableTargets.length` into a variable and use that:

https://github.com/mfdz/digitransit-ui/blob/6a8ab4c0cf9f4d6aa731b3162e93c1589f25f95a/app/component/map/tile-layer/TileLayerContainer.js#L390-L397

After that you can squash and send the PR.

**Comment by Beck-berry** [comment]
So the [PR](https://github.com/HSLdevcom/digitransit-ui/pull/3244).

**Comment by leonardehrenfried** [comment]
The PR has just been merged.

The task now is to merge upstream and integrate the refactoring into our code base.
